### ISSUE TITLE
[suggestion] imporve the search function

### ISSUE THREAD

**Issue by georgew21** [elgatito/plugin.video.elementum#664]
Hello, I would like to suggest a new way for  the search dialog, which I think is faster and better.

In providers addon, for example on elementum burst, there will be the function to organize providers on groups. So, in group 1 user will set one or two providers (these with the best results) and on group 2 he can set more providers, if the providers on group 1 don’t have the right results or don’t work etc. When elementum send command to burst to search for torrents, burst will search with group 1 of providers.

On search results dialog, except of results, would be 2 buttons: _a. “search more”_, which will give command to elementum burst to search with “group 2 of providers” and _b. “custom search query”_, so user can modify the search term (for shows with strange characters like ‘,/ “or make a search with original’s language title instead english etc) and make new search in providers of group 1. This dialog can be like subtitles search dialog.

So, searches can be fasters, on slow devices too, as for 90% of searches one provider can give the right results, but if it fails then user can quickly use group 2 of providers for more results. Also, with this way,  we will have the option to modify the search queries easily if there are problems with the titles etc.

Furthermore, I suggest to there i_s one check option for download_. If for example all the results have very few seeders or low speed, user can “check the download option” and then selecting a torrent instead of starting the playback, it will be downloaded to “files” (even if the default storage is memory).

**Comment by elgatito** [comment]
@georgew21 

> In providers addon, for example on elementum burst, there will be the function to organize providers on groups. So, in group 1 user will set one or two providers (these with the best results) and on group 2 he can set more providers, if the providers on group 1 don’t have the right results or don’t work etc. When elementum send command to burst to search for torrents, burst will search with group 1 of providers.

That needs interaction between plugins. Currently Elementum only sends request to search for something. 

> On search results dialog, except of results, would be 2 buttons: a. “search more”, which will give command to elementum burst to search with “group 2 of providers” and b. “custom search query”, so user can modify the search term (for shows with strange characters like ‘,/ “or make a search with original’s language title instead english etc) and make new search in providers of group 1. This dialog can be like subtitles search dialog.

This needs to modify search results window, add there somehow new options, process them, etc..

> So, searches can be fasters, on slow devices too, as for 90% of searches one provider can give the right results, but if it fails then user can quickly use group 2 of providers for more results. Also, with this way, we will have the option to modify the search queries easily if there are problems with the titles etc.

It will not be faster. Searching for something, then waiting for user, then searching again. That is slower than just search in parallel for 10 providers.

Kodi is not an OS, it does not allow much GUI options. Anything other than usual inputs - require a lot of work.

All above needs huge amount of time to make. Personally I, don't see any gains and think it will make search harder and slower, so I won't do that. But, again, code is open and anyone can do any changes.

**Comment by georgew21** [comment]
okay! taking granted that these are huge changes and you don't condider to make the thing fasters, i find reasonable to not make them. have nice holidays!

**Comment by antonsoroko** [comment]
>b. “custom search query”, so user can modify the search term (for shows with strange characters like ‘,/ “or make a search with original’s language title instead english etc)

btw, some time ago this was implemented via extended menu addon - https://github.com/elgatito/context.elementum
